internal/cdn/gcs_direct.go

```diff
@@ -0,0 +1,21 @@
+package cdn
+
+import (
+    "expo-open-ota/config"
+    "expo-open-ota/internal/services"
+    "fmt"
+    "os"
+    "time"
+)
+
+type GCSDirectCDN struct{}
+
+func (c *GCSDirectCDN) isCDNAvailable() bool {
+    return config.GetEnv("STORAGE_MODE") == "gcs" && config.GetEnv("GCS_BUCKET_NAME") != "" && os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") != ""
+}
+
+func (c *GCSDirectCDN) ComputeRedirectionURLForAsset(branch, runtimeVersion, updateId, asset string) (string, error) {
+    bucket := config.GetEnv("GCS_BUCKET_NAME")
+    key := fmt.Sprintf("%s/%s/%s/%s", branch, runtimeVersion, updateId, asset)
+    return services.GCSSignedURL(bucket, key, "GET", "", 15*time.Minute)
```

Comment:
Here the signed URL is set to be valid for 15mins. I'm pretty sure this will break in case of introducing a cache. Would to hear your opinion.